src/multibase.js

```diff
@@ -0,0 +1,127 @@
+'use strict'
+
+const constants = require('./constants')
+const bs58 = require('bs58')
+
+exports = module.exports = multibase
+exports.encode = encode
+exports.decode = decode
+exports.isEncoded = isEncoded
+
+const errNotSupported = new Error('Unsupported encoding')
+
+// returns a new buffer with the multibase varint+code`
+function multibase (nameOrCode, buf) {
+  if (!buf) {
+    throw new Error('requires an encoded buffer')
+  }
+  const code = getCode(nameOrCode)
+  const codeBuf = new Buffer(code)
+
+  const name = getName(nameOrCode)
+  validEncode(name, buf)
+  return Buffer.concat([codeBuf, buf])
+}
+
+function encode (nameOrCode, buf) {
+  const name = getName(nameOrCode)
+
+  let encode
+
+  switch (name) {
```

Comment:
Depends on the implementation of the base encoder/decoder or if we want to shim them beforehand to fulfil always the same interface, which is kind of what is already happening